app/data_models/questionnaire_store.py

```diff
@@ -1,23 +1,111 @@
 from __future__ import annotations
 
+from dataclasses import dataclass, field
 from datetime import datetime
-from typing import TYPE_CHECKING, MutableMapping, Optional
+from typing import TYPE_CHECKING, Iterable, MutableMapping, Optional
+
+from ordered_set import OrderedSet
 
 from app.data_models.answer_store import AnswerStore
 from app.data_models.list_store import ListStore
 from app.data_models.metadata_proxy import MetadataProxy
 from app.data_models.progress_store import ProgressStore
 from app.data_models.supplementary_data_store import SupplementaryDataStore
+from app.questionnaire import (
+    Location,
+    QuestionnaireSchema,
+    placeholder_renderer,
+    value_source_resolver,
+)
+from app.questionnaire.relationship_location import RelationshipLocation
+from app.questionnaire.rules import rule_evaluator
 from app.questionnaire.rules.utils import parse_iso_8601_datetime
 from app.utilities.json import json_dumps, json_loads
-from app.utilities.types import SupplementaryDataListMapping
+from app.utilities.types import LocationType, SupplementaryDataListMapping
 
 if TYPE_CHECKING:
     from app.storage.encrypted_questionnaire_storage import (  # pragma: no cover
         EncryptedQuestionnaireStorage,
     )
 
 
+@dataclass
+class DataStores:
+    # self.metadata is a read-only view over self._metadata
+    metadata: MetadataProxy | None = None
+    response_metadata: MutableMapping = field(default_factory=dict)
+    list_store: ListStore = ListStore()
```

Comment:
We should avoid mutable default arguments, these need the default factory too, or a normal class with none defaults

Reply:
Ok, added now, thanks.